base/dict.jl

```diff
@@ -130,10 +130,8 @@ copy(d::Dict) = Dict(d)
 
 const AnyDict = Dict{Any,Any}
 
-Dict(ps::Pair{K,V}...)           where {K,V} = Dict{K,V}(ps)
-Dict(ps::Pair{K}...)             where {K}   = Dict{K,Any}(ps)
-Dict(ps::(Pair{K,V} where K)...) where {V}   = Dict{Any,V}(ps)
-Dict(ps::Pair...)                            = Dict{Any,Any}(ps)
+Dict(ps::Pair{K,V}...) where {K,V} = Dict{K,V}(ps)
+Dict(ps::Pair...)                  = Dict(ps)
```

Comment:
Thanks! I have one question - is the first method is needed only for performance reasons or covers some special case?

Reply:
I think it's was there for performance, but that's not clear to me...